### DISCUSSION TITLE
Allow more Cache-Control settings like s-maxage or CDN specific caches

### DISCUSSION THREAD

**Discussion by MarkusWendorf** [protomaps/PMTiles#404] [Ideas]
Hey there,

I was thinking about setting the Cache-Control header to `max-age=86400, s-maxage=31536000`. 

Meaning:
- Cache the tile in the browser (end user) for 1 day
- Cache the tile in the CDN cache for 365 days

However right it is only possible to set `max-age=n` via `process.env.CACHE_MAX_AGE`. 

Would it make sense to allow set the `Cache-Control` header directly? Something like `process.env.CACHE_CONTROL`.

I would be open to implement this feature. I'm not sure if we should keep the old option and make it backwards compatible.

Thoughts?

**Comment by bdon** [comment]
What about a second configuration like `process.env.CACHE_S_MAXAGE`? Though I do agree that if there are even more options people want to customize then setting the `CACHE_CONTROL` as a single string is much more flexible. 

It should  be OK to break backwards compatibility right now since the CDN integration code isn't versioned (not sure the best approach here either)

**Reply by MarkusWendorf** [reply]
I would prefer `CACHE_CONTROL` over `CACHE_S_MAXAGE`. It might be useful for some people to use `stale-while-revalidate` for example. 

And I don't see any advantage in being over restricted here.
